hello_world sample instructions don't work

**Describe the bug**
Following the instructions in https://docs.zephyrproject.org/latest/samples/hello_world/README.html will result in a build error.

**To Reproduce**
Steps to reproduce the behavior:
1. ``cd samples/hello_world``
2. ``west build -t run``

```
-- west build: build configuration:
       source directory: /home/dbkinder/zephyrproject/zephyr/samples/hello_world
       build directory: /home/dbkinder/zephyrproject/zephyr/samples/hello_world/build (created)
       BOARD: UNKNOWN
WARNING: This looks like a fresh build and BOARD is unknown; so it probably won't work. To fix, use --board=<your-board>.
Note: to silence the above message, run 'west config build.board_warn false'
-- west build: generating a build system
Zephyr version: 2.0.0
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.6.8", minimum required is "3.4")
CMake Error at /home/dbkinder/zephyrproject/zephyr/cmake/app/boilerplate.cmake:176 (message):
  BOARD is not being defined on the CMake command-line in the environment or
  by the app.
Call Stack (most recent call first):
  CMakeLists.txt:5 (include)


-- Configuring incomplete, errors occurred!
ERROR: command exited with status 1: /home/dbkinder/.local/bin/cmake -B/home/dbkinder/zephyrproject/zephyr/samples/hello_world/build -S/home/dbkinder/zephyrproject/zephyr/samples/hello_world -GNinja
```

**Expected behavior**
Instructions should work as written.

I think what's happened here is that sometime in the past, the default board target was ``qemu_x86`` if no board was specified.  That's not the case now, and the board must be explicitly defined:

This command does run successfully:
```
west build -t run -b qemu_x86
```
But then there are no instructions for how to exit qemu :)    (CTRL-A X)

--- a/doc/extensions/zephyr/application.py
+++ b/doc/extensions/zephyr/application.py
@@ -251,27 +251,38 @@ class ZephyrAppCommandsDirective(Directive):
         if cd_into and app:
             content.append('cd {}'.format(app))
 
-        if 'build' in goals:
-            build_args = ' -b {}{}{}{}'.format(board, dst, src, cmake_args)
-            content.append('west build{}'.format(build_args))
+        # We always have to run west build.
+        #
+        # FIXME: doing this unconditionally essentially ignores the
+        # maybe-skip-config option if set.
+        #
+        # This whole script and its users from within the
+        # documentation needs to be overhauled now that we're
+        # defaulting to west.
+        #
+        # For now, this keeps the resulting commands working.
+        content.append('west build -b {}{}{}{}'.
+                       format(board, dst, src, cmake_args))
 
-        goal_args = '{}'.format(dst)
+        # If we're signing, we want to do that next, so that flashing
+        # etc. commands can use the signed file which must be created
+        # in this step.
         if 'sign' in goals:
-            content.append('west sign{}'.format(goal_args))
+            content.append('west sign{}'.format(dst))
 
         for goal in goals:
             if goal == 'build' or goal == 'sign':
                 continue
             elif goal == 'flash':
-                content.append('west flash{}'.format(goal_args))
+                content.append('west flash{}'.format(dst))
             elif goal == 'debug':
-                content.append('west debug{}'.format(goal_args))
+                content.append('west debug{}'.format(dst))
             elif goal == 'debugserver':
-                content.append('west debugserver{}'.format(goal_args))
+                content.append('west debugserver{}'.format(dst))
             elif goal == 'attach':
-                content.append('west attach{}'.format(goal_args))
+                content.append('west attach{}'.format(dst))
             else:
-                content.append('west build -t {}{}'.format(goal, goal_args))
+                content.append('west build -t {}{}'.format(goal, dst))
 
         return content
 

